airbyte_cdk/sources/streams/concurrent/helpers.py

```diff
@@ -13,8 +13,13 @@ def get_primary_key_from_stream(
     elif isinstance(stream_primary_key, str):
         return [stream_primary_key]
     elif isinstance(stream_primary_key, list):
-        if len(stream_primary_key) > 0 and all(isinstance(k, str) for k in stream_primary_key):
+        are_all_elements_str = all(isinstance(k, str) for k in stream_primary_key)
+        are_all_elements_list_of_size_one = all(isinstance(k, list) and len(k) == 1 for k in stream_primary_key)
```

Comment:
Should we also verify that each list also contains a string?

`are_all_elements_list_of_size_one_and_str = all(isinstance(k, list) and len(k) == 1 and all(isinstance(j, str) for j in k) for k in test)`

Reply:
> covered by the typing of stream_primary_key: Optional[Union[str, List[str], List[List[str]]]]